src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -237,6 +239,10 @@ private void releaseExecutor(Execution execution) {
 			LOG.error("Released execution was not found in collection of executions currently being processed. Should never happen.");
 			statsRegistry.registerUnexpectedError();
 		}
+
+		if(pollAfterCompletion) {
```

Comment:
Just a thought, we might also check here if we got a full batch of executions, if not there is not need to check again